### PR TITLE
Bip32

### PR DESCRIPTION
**Pull request by valentunn**
# Warning
Not compatible changes with the previous version - the entire crypto layer is rewritten to provide more modularity and easy-to-use API.

### CODE REVIEW

File: fearless-utils/src/test/java/jp/co/soramitsu/fearless_utils/encrypt/seed/ethereum/EthereumSeedFactoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import com.google.gson.Gson
+import jp.co.soramitsu.fearless_utils.encrypt.keypair.ethereum.EthereumKeypairFactory
+import jp.co.soramitsu.fearless_utils.extensions.toHexString
+import jp.co.soramitsu.fearless_utils.getResourceReader
+import jp.co.soramitsu.fearless_utils.encrypt.junction.BIP32JunctionDecoder
+import jp.co.soramitsu.fearless_utils.encrypt.mnemonic.MnemonicTestCase
+import org.junit.Assert
+import org.junit.Test
+
+class EthereumSeedFactoryTest {
```
Comment (stas-litvinov) [inline fearless-utils/src/test/java/jp/co/soramitsu/fearless_utils/encrypt/seed/ethereum/EthereumSeedFactoryTest.kt:12]: Looks like it is not exactly seed factory test — it is keypair generator test. Maybe, it's better give it more concise name

File: fearless-utils/src/testShared/java/jp/co/soramitsu/fearless_utils/encrypt/mnemonic/SubstrateSeedFactoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import com.google.gson.Gson
+import jp.co.soramitsu.fearless_utils.encrypt.EncryptionType
+import jp.co.soramitsu.fearless_utils.encrypt.keypair.substrate.SubstrateKeypairFactory
+import jp.co.soramitsu.fearless_utils.extensions.toHexString
+import jp.co.soramitsu.fearless_utils.getResourceReader
+import jp.co.soramitsu.fearless_utils.encrypt.junction.SubstrateJunctionDecoder
+import jp.co.soramitsu.fearless_utils.encrypt.seed.substrate.SubstrateSeedFactory
+import org.junit.Assert
+
+abstract class SubstrateSeedFactoryTest {
```
Comment (stas-litvinov) [inline fearless-utils/src/testShared/java/jp/co/soramitsu/fearless_utils/encrypt/mnemonic/SubstrateSeedFactoryTest.kt:12]: I suggest to rename it as well to a `SubstrateKeypairGenerationTest`or something like this

**Review by stas-litvinov — APPROVED** [review]

File: fearless-utils/src/test/java/jp/co/soramitsu/fearless_utils/encrypt/seed/ethereum/EthereumSeedFactoryTest.kt (hunk 1 above)
Comment (valentunn) [reply fearless-utils/src/test/java/jp/co/soramitsu/fearless_utils/encrypt/seed/ethereum/EthereumSeedFactoryTest.kt]: Done

File: fearless-utils/src/testShared/java/jp/co/soramitsu/fearless_utils/encrypt/mnemonic/SubstrateSeedFactoryTest.kt (hunk 2 above)
Comment (valentunn) [reply fearless-utils/src/testShared/java/jp/co/soramitsu/fearless_utils/encrypt/mnemonic/SubstrateSeedFactoryTest.kt]: Done

**Pull request merged**